src/VirtoCommerce.CatalogModule.Data/Repositories/CatalogRepositoryImpl.cs

```diff
@@ -441,6 +441,16 @@ public async Task RemoveAllPropertyValuesAsync(string propertyId)
             }
         }
 
+        public async Task RemoveDynamicAssociationsAsync(string[] dynamicAssociationIds)
```

Comment:
Why do we need this method in the repository? Seems there are no specific SQL commands, maybe you could use DynamicAssociations collection for deleting directly in the service.